api_internal.go

```diff
@@ -367,9 +367,67 @@ func (c *Couchbase) uploadClusterCACert(pem []byte) error {
 	return c.n_post("/controller/uploadClusterCA", pem, nil, headers)
 }
 
+func (c *Couchbase) getClusterVersion() (v string, err error) {
+	info, err := c.getPools()
+	if err != nil {
+		return
+	}
+	arr := strings.Split(info.ImplVersion, "-")
+	v = arr[0]
+	return
+}
+
+func splitVersionString(v string) (major, minor, patch int, err error) {
+	arr := strings.Split(v, ".")
+	if len(arr) < 3 {
+		err = fmt.Errorf("version string = %s is invalid", v)
+		return
+	}
+	major, err = strconv.Atoi(arr[0])
+	if err != nil {
+		return
+	}
+	minor, err = strconv.Atoi(arr[1])
+	if err != nil {
+		return
+	}
+	patch, err = strconv.Atoi(arr[2])
+	return
+}
+
+func (c *Couchbase) getClusterCert71() (cert []byte, err error) {
```

Comment:
Not just 7.1, but post 7.1

Reply:
It's not intended only for 7.1.
If there is a new way to get certs in say 7.6 then we will have function getClusterCert76

Reply:
Understood